salt/modules/junos.py

```
@@ -150,75 +150,50 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
             conn = __proxy__["junos.conn"]()
-            restore_timeout = conn.timeout
+            old_timeout = conn.timeout
+            # Default timeout is 60
+            # the int(x or 60) solves the problem where None is provided
+            ldev_timeout = max(
+                int(kwargs.pop("dev_timeout", 0) or 60),
+                int(kwargs.pop("timeout", 0) or 60),
+            )
             conn.timeout = ldev_timeout
-            try:
-                restore_kwargs = False
-                del_list = []
-                op = {}
-                op.update(kwargs)
-                for keychk in kwargs:
-                    if keychk.startswith("__pub"):
-                        del_list.append(keychk)
-                if del_list:
-                    restore_kwargs = True
-                    for delkey in del_list:
-                        kwargs.pop(delkey)
-
-                result = function(*args, **kwargs)
-                if restore_kwargs:
-                    kwargs.update(op)
-
-                conn.timeout = restore_timeout
-                return result
-            except Exception:  # pylint: disable=broad-except
-                conn.timeout = restore_timeout
-                raise
-        else:
-            restore_kwargs = False
-            del_list = []
-            op = {}
-            op.update(kwargs)
-            for keychk in kwargs:
-                if keychk.startswith("__pub"):
-                    del_list.append(keychk)
-            if del_list:
-                restore_kwargs = True
-                for delkey in del_list:
-                    kwargs.pop(delkey)
-
-            ret = function(*args, **kwargs)
-            if restore_kwargs:
-                kwargs.update(op)
+            timeout_changed = True
 
-            return ret
+        # Run the function
+        try:
+            result = function(*args, **kwargs)
+        except:
+            raise
```

```suggestion
```
Don't catch exceptions just to raise them afterwards without doing anything with that information.

---

It was done like that before therefor I keept the same functionality.

---

I have to disagree.
Neither [this](https://github.com/saltstack/salt/blob/3006.x/salt/modules/junos.py#L153-L169) chunk, nor [this](https://github.com/saltstack/salt/blob/3006.x/salt/modules/junos.py#L174-L203) has an exception caught just to be raised immediately.
There's code that runs, before raising the exception again.

Please remove that bare except, you're not doing anything with it.